client/secureclient/src/main/java/com/linkedin/openhouse/client/ssl/WebClientFactory.java

```diff
@@ -205,6 +197,18 @@ private void setSessionIdInWebClientHeader(WebClient.Builder webClientBuilder) {
     webClientBuilder.defaultHeaders(h -> h.add(SESSION_ID, sessionId));
   }
 
+  /**
+   * Set clientName in the header of webClient. If a clientName is not provided, do nothing.
+   *
+   * @param webClientBuilder
+   */
+  private void setClientNameInWebClientHeader(WebClient.Builder webClientBuilder) {
+    if (clientName != null) {
+      webClientBuilder.defaultHeaders(h -> h.add(HTTPHEADER_CLIENT_NAME, clientName));
+    }
```

Comment:
I think this is the right choice, to leave this client_name_header as null when clientName is not available from client-side, since we will live with a mix of client that doesn't have / have this field for a while ? 
But accordingly have we fully tested server behavior for this ?